server/enhancedDate.js

```diff
@@ -1 +1,103 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+
+
+'use strict';
+var dateObjects = {
+  dateTime: '',
+  monthNames: ['January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'],
+  dayNames: ['Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday'],
+  milsDay: 86400000
+};
+
+var now = function () {
+  return new Date();
+};
+
+var setDate = function (e) {
+    if (e instanceof Date) {
+      dateObjects.dateTime = e;
+    }
+    else if (!isNaN(e)) {
+      dateObjects.dateTime = new Date(e);
+    }
+    else if (e === undefined) {
+      dateObjects.dateTime = new Date();
+    }
+    else {
+      console.warn('argument is not instanceof Date or epoch');
+    }
+    return dateObjects.dateTime;
+};
+
+var getDate = function (e) {
+    var time = dateObjects.dateTime;
+
+    // set current time if dateObject.date is undefined
+    if (time === undefined || !(time instanceof Date))  {
+      dateObjects.dateTime = now();
+      time = dateObjects.dateTime;
+    }
+
+    if (e !== true) {
+      time = dateObjects.dateTime.getTime();
+    }
+    return time;
+};
+
+var getDayName = function () {
+  var day = dateObjects.dateTime.getDay();
+  return dateObjects.dayNames[day];
+};
+
+var getMonthName = function () {
+  var month = dateObjects.dateTime.getMonth();

```

Comment:
This function should check if dateObjects.dateTime has been set first and if it hasn't, create a new Date and set it. By trying to access methods on an object that hasn't been assigned anything yet, this code won't work unless the user explicitly calls setDate first.